File: week14/vblog/api/apps/blog/impl/blog.go
package impl

import (
	"context"
	"go_8_mage/week14/vblog/api/apps/blog"
)

//创建文章
func (i *Impl)CreateBlog(ctx context.Context, req *blog.CreateBlogRequest) (*blog.Blog, error){
	return nil,nil
}

//	更新文章
func (i *Impl)UpdateBlog(ctx context.Context, req *blog.UpdateBlogRequest) (*blog.Blog, error){
	return nil,nil
}

//	文章的删除
func (i *Impl)DeleteBlog(ctx context.Context, req *blog.DeleteBlogRequest) (*blog.Blog, error){
	return nil,nil
}

//	文章列表
func (i *Impl)QueryBlog(ctx context.Context, req *blog.QueryBlogRequest) (*blog.BlogSet, error){
	return nil,nil
}

//	文章详情
func (i *Impl)DescribeBlog(ctx context.Context, req *blog.DescribeBlogRequest) (*blog.Blog, error){
	return nil,nil
}

//	发布接口
func (i *Impl)UpdateBlogStatus(ctx context.Context, req *blog.UpdateBlogStatusRequest) (*blog.Blog, error){
	return nil,nil
}

File: week14_after/vblog/ui/src/api/blog.js
import client from "./client";

export function LIST_BLOG(params) {
    return client({
        url: "/vblog/api/v1/blog/",
        method: "get",
        params: params,
    });
}

export function GET_BLOG(id, params) {
    return client({
        url: `/vblog/api/v1/blog/${id}`,
        method: "get",
        params: params,
    });
}

export function DELETE_BLOG(id, params) {
    return client({
        url: `/vblog/api/v1/blog/${id}`,
        method: "delete",
        params: params,
    });
}

export function CREATE_BLOG(data) {
    return client({
        url: `/vblog/api/v1/blog`,
        method: "post",
        data: data,
    });
}
export function UPDATE_BLOG(id,data) {
    return client({
        url: `/vblog/api/v1/blog/${id}`,
        method: "put",
        data: data,
    });
}

export function UPDATE_BLOG_STATUS(id,data) {
    return client({
        url: `/vblog/api/v1/blog/${id}/status`,
        method: "post",
        data:data,
    });
}

